src/main/java/org/sagebionetworks/bridge/spring/controllers/HealthDataDocumentationController.java

```diff
@@ -0,0 +1,106 @@
+    @PostMapping(path="/v3/healthdataDocumentation")
+    public HealthDataDocumentation createOrUpdateHealthDataDocumentation() {
+        UserSession session = getAuthenticatedSession(Roles.RESEARCHER, Roles.DEVELOPER);
+
+        HealthDataDocumentation documentation = parseJson(HealthDataDocumentation.class);
+
+        if (documentation.getDocumentation() == null) {
+            throw new BadRequestException("Documentation is required to store health data documentation.");
+        }
+
+        // check documentation size (max allowed is 100 KB)
+        if (documentation.getDocumentation().getBytes(Charset.defaultCharset()).length > MAX_DOCUMENTATION_BYTES) {
+            throw new BadRequestException("Documentation must be less than 100 KB");
+        }
+
+        // update health data documentation attributes
+        if (documentation.getCreatedOn() == null) {
+            documentation.setCreatedOn(DateTime.now());
+            documentation.setCreatedBy(session.getId());
+        } else {
+            documentation.setModifiedOn(DateTime.now());
+            documentation.setModifiedBy(session.getId());
+        }
+
+        return healthDataDocumentationService.createOrUpdateHealthDataDocumentation(documentation);
+    }
+
+    /** Get a health data documentation with the given identifier. */
+    @GetMapping(path="/v3/healthdataDocumentation/{identifier}")
+    public HealthDataDocumentation getHealthDataDocumentationForId(@PathVariable String identifier) {
+        UserSession session = getAuthenticatedSession(Roles.RESEARCHER, Roles.DEVELOPER);
+
+        // AppId is a placeholder for parentId. In the future, for study-scoped documentation, it would be “{appId}:study:{studyId}”
+        // and for Assessments it would be “{appId}:assessment:{assessmentId}”.
+        String parentId = session.getAppId();
+        return healthDataDocumentationService.getHealthDataDocumentationForId(identifier, parentId);
+    }
+
+    /** Get all health data documentation with the given parentId. */
+    @GetMapping(path="/v3/healthdataDocumentation")
+    public ForwardCursorPagedResourceList<HealthDataDocumentation> getAllHealthDataDocumentationForParentId(@RequestParam String parentId,
+        @RequestParam(required = false) String pageSize, @RequestParam(required = false) String offsetKey) {
+        getAuthenticatedSession(Roles.RESEARCHER, Roles.DEVELOPER);
+
+        int pageSizeInt = BridgeUtils.getIntOrDefault(pageSize, API_DEFAULT_PAGE_SIZE);
+
+        return healthDataDocumentationService.getAllHealthDataDocumentation(parentId, pageSizeInt, offsetKey);
+    }
+
+    /** Delete a health data documentation with the given identifier. */
+    @DeleteMapping(path="v3/healthdataDocumentation/{identifier}")
+    public StatusMessage deleteHealthDataDocumentationForIdentifier(@PathVariable String identifier) {
+        UserSession session = getAuthenticatedSession(Roles.RESEARCHER, Roles.DEVELOPER);
```

Comment:
Thinking through this some more, I think the delete APIs should be admin only. This way, you can't accidentally delete someone else's documentation, or worse, all the documentation in an app.